src/parser/expression.js

```diff
@@ -786,6 +795,8 @@ export default class ExpressionParser extends LValParser {
     node.callee = this.parseNoCallExpr();
     const optional = this.eat(tt.questionDot);
 
+    this.parseNewTypeArguments(node);
```

Comment:
I don't think the AST gives us a way to distinguish between `new X` and `new X()`. But I found a different way that works just as well, see the latest commit.